From 9b092da5a508ba119747ca6ce5ed84d030414989 Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Mon, 17 Jan 2011 10:40:13 +0800
Subject: [PATCH 111/635] ENGR00137877 [Android_imx51_BBG] Camera: open
 camera/camcorder fail

The root cause is that the system updated, so the sem_init changed and
the api changed.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 libcamera/CameraHal.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 205b986..a528958 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -118,7 +118,7 @@ void CameraHal::initDefaultParameters()
 
     p.setPreviewSize(RECORDING_WIDTH_NORMAL, RECORDING_HEIGHT_NORMAL);
     p.setPreviewFrameRate(PREVIEW_FRAMERATE);
-    p.setPreviewFormat("yuv420sp");
+    p.setPreviewFormat(CameraParameters::PIXEL_FORMAT_YUV420SP);
 
     p.setPictureSize(PICTURE_WIDTH, PICTURE_HEIGHT);
     p.setPictureFormat("jpeg");
@@ -142,7 +142,7 @@ void CameraHal::initDefaultParameters()
     p.set(CameraParameters::KEY_SUPPORTED_JPEG_THUMBNAIL_SIZES, CameraHal::supprotedThumbnailSizes);
 	p.set(CameraParameters::KEY_JPEG_THUMBNAIL_WIDTH, "96");
 	p.set(CameraParameters::KEY_JPEG_THUMBNAIL_HEIGHT, "96");
-
+	p.set(CameraParameters::KEY_VIDEO_FRAME_FORMAT, CameraParameters::PIXEL_FORMAT_YUV420SP);
 
     memset(tmpBuffer, '\0', PARAM_BUFFER);
     strncat((char*) tmpBuffer, (const char*) CameraParameters::WHITE_BALANCE_AUTO, PARAM_BUFFER);
-- 
1.8.0

